v2/dtos/intervalaction.go

```diff
@@ -0,0 +1,94 @@
+//
+// Copyright (C) 2021 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package dtos
+
+import (
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2/dtos/common"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2/models"
+)
+
+// IntervalAction and its properties are defined in the APIv2 specification:
+// https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/support-scheduler/2.x#/IntervalAction
+type IntervalAction struct {
+	common.Versionable `json:",inline"`
+	Created            int64  `json:"created,omitempty"`
+	Modified           int64  `json:"modified,omitempty"`
+	Id                 string `json:"id,omitempty" validate:"omitempty,uuid"`
+	Name               string `json:"name" validate:"edgex-dto-none-empty-string,edgex-dto-rfc3986-unreserved-chars"`
+	IntervalName       string `json:"intervalName" validate:"edgex-dto-none-empty-string,edgex-dto-rfc3986-unreserved-chars"`
+	Protocol           string `json:"protocol,omitempty"`
+	Host               string `json:"host,omitempty"`
+	Port               int    `json:"port,omitempty"`
+	Path               string `json:"path,omitempty"`
+	Parameters         string `json:"parameters,omitempty"`
+	HTTPMethod         string `json:"httpMethod,omitempty" validate:"omitempty,oneof='GET' 'HEAD' 'POST' 'PUT' 'DELETE' 'TRACE' 'CONNECT'"`
+	User               string `json:"user,omitempty"`
+	Password           string `json:"password,omitempty"`
```

Comment:
@cloudxxx8 , @jpwhitemn , @hutchic , Looks like have need for Scheduler to store these secrets in Vault.

Reply:
@lenny-intel we need re-design the interval-action.  V1 doesn't support MQTT, but there are lots of message bus fields here.  I prefer to support pure REST without security in the initial version.

Reply:
@cloudxxx8 , thanks for the clarification. I agree with REST only until we have a use case that requires MQTT.